subsys/mgmt/hawkbit/hawkbit.c

```diff
@@ -60,17 +60,13 @@ static bool hawkbit_initialized;
 
 #ifndef CONFIG_HAWKBIT_DDI_NO_SECURITY
 
-#ifdef CONFIG_HAWKBIT_DDI_GATEWAY_SECURITY
-#define AUTH_HEADER_START "Authorization: GatewayToken "
-#else
-#define AUTH_HEADER_START "Authorization: TargetToken "
-#endif /* CONFIG_HAWKBIT_DDI_GATEWAY_SECURITY */
+#define AUTH_HEADER_START                                                                          \
+	"Authorization: " COND_CODE_1(CONFIG_HAWKBIT_DDI_GATEWAY_SECURITY, ("Gateway"),            \
+				      ("Target")) "Token "
 
-#ifdef CONFIG_HAWKBIT_SET_SETTINGS_RUNTIME
-#define AUTH_HEADER_FULL AUTH_HEADER_START "%s" HTTP_CRLF
-#else
-#define AUTH_HEADER_FULL AUTH_HEADER_START CONFIG_HAWKBIT_DDI_SECURITY_TOKEN HTTP_CRLF
-#endif /* CONFIG_HAWKBIT_SET_SETTINGS_RUNTIME */
+#define AUTH_HEADER_FULL                                                                           \
```

Comment:
Yes, it is shorter now, but I'm not sure if this made it easier to read. I'd rather have a few more lines in the source file than something that's a few lines shorter but harder to read..